xonsh/aliases.py

```diff
@@ -73,6 +73,19 @@ def source_bash(args, stdin=None):
     return
 
 
+def exec(args, stdin=None):
```

Comment:
If you are just looking for a new name, `xexec` and `exec-xonsh` are both free.  The more that I think about it, the more that a new (alias) name would be needed, even if this was a thin wrapper around execx.